pageserver/src/tenant/timeline.rs

```diff
@@ -979,9 +979,13 @@ impl Timeline {
     /// Calculate the logical size of the database at the latest LSN.
     ///
     /// NOTE: counted incrementally, includes ancestors, this can be a slow operation.
-    fn calculate_logical_size(&self, up_to_lsn: Lsn) -> anyhow::Result<u64> {
+    pub fn calculate_logical_size(&self, up_to_lsn: Lsn) -> anyhow::Result<u64> {
         info!("Calculating logical size for timeline {}", self.timeline_id);
-        let timer = self.metrics.init_logical_size_histo.start_timer();
+        let timer = if up_to_lsn == self.initdb_lsn {
```

Comment:
> Is it important to distinguish such cases? 

Distinguishing: I was thinking of it from the perspective of the metrics we've gathered 
and assume that the later LSN queries done will be much more expensive.

> that are created but never updated that would have `up_to_lsn == self.initdb_lsn` condition always true

I don't think the `initdb_lsn` is currently being queried because I crudely filter out any `gc_cutoff` in the parent branch, which might require fixing later but right now it's difficult to say if it's right or wrong. I didn't test this out at `initdb_lsn` because that would be below gc_horizon I guess in all usual configurations..

Even if this was true, it would be calculated one extra time per tenant only.

Idea of using the `Timeline::current_logical_size` as cache if it has already been calculated makes sense though. Should it be done on this PR?

Reply:
Depends how simple it is to do 🙂 
I still insist on the incremental, segment-based calculations for timeline sizes later, so this could be implemented later together.

Reply:
Had to rebase, but I think it's simple enough: 36f035d4f8804f7922da435baa7ef8f3ce58915b, what do you think?